tree/ntupleutil/v7/inc/ROOT/RNTupleImporter.hxx

```diff
@@ -16,41 +16,238 @@
 #ifndef ROOT7_RNTuplerImporter
 #define ROOT7_RNTuplerImporter
 
+#include <ROOT/REntry.hxx>
+#include <ROOT/RError.hxx>
+#include <ROOT/RField.hxx>
+#include <ROOT/RNTuple.hxx>
+#include <ROOT/RNTupleModel.hxx>
+#include <ROOT/RNTupleOptions.hxx>
 #include <ROOT/RStringView.hxx>
 
+#include <TFile.h>
+#include <TTree.h>
+
+#include <cstdlib>
+#include <map>
 #include <memory>
+#include <vector>
 
-class TTree;
+class TLeaf;
 
 namespace ROOT {
 namespace Experimental {
 
-namespace Detail {
-class RPageSink;
-}
-
 // clang-format off
 /**
 \class ROOT::Experimental::Detail::RNTupleImporter
 \ingroup NTuple
 \brief Converts a TTree into an RNTuple
 
-The class steers the conversion of a TTree into an RNTuple.
+Example usage:
+
+~~~ {.cpp}
+#include <ROOT/RNTupleImporter.hxx>
+using ROOT::Experimental::RNTupleImporter;
+
+auto importer = RNTupleImporter::Create("data.root", "TreeName", "output.root").Unwrap();
+// As required: importer->SetNTupleName(), importer->SetWriteOptions(), ...
+importer->Import().ThrowOnError();
+~~~
+
+The output file is created if it does not exist, otherwise the ntuple is added to the existing file.
+Note that input file and output file can be identical if the nutple is stored under a different name than the tree
+(use `SetNTupleName()`).
+
+By default, the RNTuple is compressed with zstd, independent of the input compression. The compression settings
+(and other output parameters) can be changed by `SetWriteOptions()`.
+
+Most RNTuple fields have a type identical to the corresponding TTree input branch. Exceptions are
+  - C string branches are translated to std::string fields
+  - C style arrays are translated to std::array<...> fields
+  - Leaf lists are translated to untyped records
+  - Leaf count arrays are translated to anonymous collections with generic names (_collection0, collection1, etc.).
+    In order to keep field names and branch names aligned, RNTuple projects the members of these collections and
+    its collection counter to the input branch names. For instance, the input leafs
+      Int_t njets;
+      float jet_pt[njets]
+      float jet_eta[njets]
+    will be converted to the following RNTuple schema:
+      _collection0 (untyped collection)
+      |- float jet_pt
+      |- float jet_eta
+      std::uint64_t njets (projected from _collection0 without subfields)
+      ROOT::RVec<float> jet_pt (projected from _collection0.jet_pt)
+      ROOT::RVec<float> jet_eta (projected from _collection0.jet_eta)
+    These projections are meta-data only operations and don't involve duplicating the data.
```

Comment:
Are 'projections' akin to aliases?

Reply:
Possibly. Are "typed aliases" currently a thing?  Projections are more than just renaming existing fields.  A projected field combines existing columns to a new type. In the case of `jet_pt`, the field takes the offset column from `_collection0` and the float column from `_collection0.jet_pt` and creates a new type from the two, `RVec<float>`.

Reply:
> Are "typed aliases" currently a thing?  

No :).  I am distinguishing it from the string-only alias in `TTree::Draw`

Thanks for the clarification.